src/main/java/org/gridsuite/odre/server/utils/GeographicDataParser.java

```diff
@@ -82,7 +82,34 @@ private GeographicDataParser() {
         return substations;
     }
 
-    public static Map<String, LineGeoData> parseLines(BufferedReader aerialLinesBr, BufferedReader undergroundLinesBr) {
+    private static double distanceCoordinate(Coordinate coord1, Coordinate coord2) {
+        return DistanceCalculator.distance(coord1.getLat(), coord1.getLon(), coord2.getLat(), coord2.getLon());
+    }
+
+    public static String findSubstationStart(Map<String, SubstationGeoData> substationGeoData, String lineId, List<Coordinate> coordinates) {
+        String substation1 = lineId.substring(0, 5).trim();
```

Comment:
Maybe add an error log when lineId.substring(5,8) is not the expected thing. For exemple L76 or L63 is valid.

Reply:
how do i know that an id is invalid ? Some stations have 3 characters, and space, others have number...

Reply:
In the dataset, all lines have aaaaaLXYbbbbb no ? where X and Y are numbers between 1 and 7 ? just check L, X and Y ?

Reply:
> Some stations have 3 characters

are you sure ? how does this work with `lineId.substring(0, 5)` and after that `lineId.substring(8)` ?